src/_posts/2020-06-01-signs-your-software-is-rotting.md

```diff
@@ -0,0 +1,125 @@
+
+## Fragility ##
+
+Fragility refers to software that tends to break in many places whenever a change is made, often even in areas that are conceptually unrelated to the change being made. As this increases, the software becomes very difficult to maintain because every new change introduces numerous new defects. In the best case, these defects are caught early by an automated testing suite. Worse they are found in production by end-users. This fragility can then lead to a loss of credibility for the software and the team owning it. 
+
+## Time to deliver features increases ##
+
+A particularly strong indicator of software rot is when a team starts to see the time to deliver new features to a codebase increase. An ever increasing amount of time needed to add new features to a codebase is a sign of code rigidity. This effectively means code that is difficult to change, more often than not because code is tightly coupled. For example, a new change causes a cascade of subsequent changes in dependent modules with the codebase. This results in teams often being fearful to address non-critical problems because they do not know the full impact of making one change, or how long that change will take. 
+
+## Struggling to keep pace with the demands of the business ##
+
+Closely linked to the previous point, when a team starts to struggle to keep pace with the demands of the organisation that is also an indication that the software codebase could be in an unhealthy state. For example, perhaps the domain modelling in the codebase is no longer a good fit and requires a new model to be designed in order to facilitate the needs of the business requirements. When a software team is unable to keep pace with the demands of the organisation there is a serious problem. It means that the organisation is at risk of losing any competitive edge it has in the market as it will take it longer to ship new features to its customers. This leaves the organisation open to be overtaken by competitors who are able to deliver new features to customers much faster and on a continuous basis.
+
+## Change Failure Rate ##
+
+It’s all well and good deploying your software to production frequently, but if these deployments result in breaking changes being introduced that impact your customer’s ability to use your product then that’s not good at all. We want to be deploying new changes into production regularly, but not at the cost of quality. Change Failure Rate is a metric that was introduced by the DevOps Research and Assessment team (DORA) and is a measure of how often a deployment to production results in a failure of some kind being introduced. A high or increasing change failure rate suggests that our software is not as healthy as it should be and is likely missing adequate quality gates on the path to production. I wrote about this in a previous article, which you can find here : https://codurance.com/2020/03/12/achieving-stability-and-speed-in-software-delivery.
```

Comment:
Last sentence in this paragraph: I wrote about this in a previous article, which you can find here : https://codurance.com/2020/03/12/achieving-stability-and-speed-in-software-delivery.

Change to:
I wrote about change failure rate in a previous article, which you can find [here](https://codurance.com/2020/03/12/achieving-stability-and-speed-in-software-delivery).